clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -81,6 +81,14 @@
     public List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Disconnect connection to a particular node (if there is one). The disconnection is asynchronous and will not be
+     * processed until the next {@link #poll(long, long) poll()} call.
```

Comment:
Yes, that's how MockSelector is implemented. In the real selector, my understanding is that if you disconnect a client, we cancel the key. The cancelled key will never be selected by the selector again. So, you would have to set the connection state in the same disconnect call. If that's the case, we probably need to change the behavior of MockSelector.